src/modules/telemetry-helper.js

```diff
@@ -1,29 +1,70 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 
 var clientSocket;
+var serviceToPluginMap;
+var pendingTelemetryEvents = [];
 
 function init(socket) {
     clientSocket = socket;
+    trySendPendingEvents();
 }
 
-function sendClientTelemetry(event, props, piiProps) {
+function registerPluginServices(pluginServices) {
+    serviceToPluginMap = pluginServices;
+    trySendPendingEvents();
+}
+
+function trySendPendingEvents() {
     if (!clientSocket) {
         return;
     }
 
-    var telemetryData = {
+    var unsent = [];
+
+    pendingTelemetryEvents.forEach(function (eventData) {
+        if (mustMapServiceToPlugin(eventData) && !serviceToPluginMap) {
+            unsent.push(eventData);
+        } else {
+            sendClientTelemetry(eventData);
+        }
+    });
+    
+    pendingTelemetryEvents = unsent;
+}
+
+function mustMapServiceToPlugin(eventData) {
+    return !!eventData.props.service && !eventData.plugin;
+}
+
+function sendClientTelemetry(event, props, piiProps) {
+    var eventData = {
         event: event,
         props: props,
         piiProps: piiProps
```

Comment:
PII is acronym for "Personally Identifiable Information". In case of telemetry, we want to separate what is "sensitive" info and what is not, so we put them in `piiProps` and `props` respectively.

Please note that cordova-simulate does not do anything special with PII data; it's up to the programmatic consumer to decide how to deal with it. In case of our cordova extension, for example, when `piiProps` is not empty, we hash all the values contained in it so that it's impossible to know what the real values are (even for us).

As to why we consider plugins to be PII, it's because Cordova supports custom plugins, which can be installed from local paths, which can expose information about the user and his machine. So, for any plugin that is not built-in, we use `piiProps` to notify the programmatic consumer that it should anonymize the properties it contains.

Hope this helps!